### DISCUSSION TITLE
Help with user photo in signature

### DISCUSSION THREAD

**Discussion by marktucker08** [Set-OutlookSignatures/Set-OutlookSignatures#122] [Q&A] [answered]
I am having trouble getting the user's photo to replace the placeholder image in the signature. All other attributes are populating. I have followed the steps in the readme file for the image. This is a fully Azure AD/Entra domain. I'm not sure what the hang up is. Attaching the verbose script output.
[verbose.txt](https://github.com/user-attachments/files/16619661/verbose.txt)

**Comment by GruberMarkus** [comment, marked as answer]
I just had a quick log at the log. The user photo is there (quite big with 2.6 MB), the templates get applied.

The problem is very likely in the template itself, specifically the way the replacement photo has been inserted into the word document.

In v4.13.0, chapter '13.1.1. When using DOCX template files' describes what to do. You have to follow these instructions strictly, especially the **'Insert and Link'** part. Without **'Insert and Link'**, image replacement does not work due to Word restrictions.

For easier understanding, I would skip the optional step 2, which implies that step 4 is no longer optional but mandatory.


Independent from the image topic:
- As you have everything in the cloud, consider using the '-GraphOnly true' parameter.
- You may not need to assing the templates to $CurrentUserMail$. Just remove that line, and the template will be assigned to every mailbox as default (with the primary mail address of the current user having the highest priority).

**Reply by marktucker08** [reply]
Hey Markus, thanks for your quick reply. I was missing correctly completing the 'Insert and Link' step. Also, thank you for your suggestions as those will make this implementation less complex. Thanks again!

**Reply by GruberMarkus** [reply]
My pleasure!
